astroid/nodes/scoped_nodes.py

```diff
@@ -77,6 +77,8 @@
     {"classmethod", "staticmethod", "builtins.classmethod", "builtins.staticmethod"}
 )
 
+T = TypeVar("T")
```

Comment:
I don't understand this part, what does ``T`` means ?

Reply:
I don't know how it works under the hood. But for
```
def scope(self: T) -> T:
        """The first parent node defining a new scope.

        :returns: The first parent scope node.
        :rtype: Module or FunctionDef or ClassDef or Lambda or GenExpr
        """
        return self
```
It makes the function return with the correct typing. It sets the return type to the type of `self`.

Reply:
Ok I've read the documentation and I think ``T = TypeVar("T")`` is the example for ``TypeVar`` from the python doc, but I guess we can change the name to something else ? Maybe ``Scope`` ?

Reply:
@cdce8p what do you think, isn't it better to have an explicit name here ?

Reply:
Sorry for the discrepancies @DanielNoord I started to not review some MR that Marc is already reviewing because there is a lot to review :)

Reply:
No worries! With the typing PRs it never hurts to have an additional set of eyes.

Reply:
It would be possible to rename the `TypeVar`, I just wouldn't recommend it in this case. `T` is the usual convention. Additionally, it doesn't have any constraints or bounds. Thus it could be reused in another function later.

That is, for example, different in the `TypeVar` we used in https://github.com/PyCQA/pylint/pull/4978.